poh-bundle-linux-x64/oh.sh

```diff
@@ -1,68 +1,196 @@
+sed -e "s/3306/$MYSQL_PORT/" $POH_PATH/$OH_DIR/rsc/database.properties.sample > $POH_PATH/$OH_DIR/rsc/database.properties
+sed -e "s/MYSQL_PORT/$MYSQL_PORT/" $POH_PATH/$OH_DIR/rsc/log4j.properties.ori > $POH_PATH/$OH_DIR/rsc/log4j.properties
 
 if [ -f $POH_PATH/database.sql ]
 then
-    echo "Initializing database... on port $mysql_port"
-    cd $POH_PATH/$MYSQL_DIR/
-    rm -rf ../var/lib/mysql
-    mkdir -p ../var/lib/mysql
-    mkdir -p ../var/log/mysql
-    ./bin/mysqld --initialize-insecure --basedir=./ --datadir=../var/lib/mysql
+    echo "Initializing MySQL database... on port $MYSQL_PORT"
+#    cd $POH_PATH/$MYSQL_DIR/
+    rm -rf $POH_PATH/var/lib/mysql
+    mkdir -p $POH_PATH/var/lib/mysql
+    mkdir -p $POH_PATH/var/log/mysql
+    $POH_PATH/$MYSQL_DIR/bin/mysqld --initialize-insecure --basedir=$POH_PATH/$MYSQL_DIR --datadir=$POH_PATH/var/lib/mysql
     if [ $? -ne 0 ]; then
-		echo "Error: Initialization failed!"
+		echo "Error: MySQL initialization failed!"
 		exit 1
 	fi
 	echo "Default schemas initialized..."
-	echo -n "Starting MySQL Server."
-	./bin/mysqld_safe --defaults-file=$POH_PATH/etc/mysql/my.cnf 2>&1 > /dev/null &
-    # Wait till the MySQL socket file is created
-    while [ ! -e $POH_PATH/var/run/mysqld/mysql.sock ]; do sleep 1; done
-    ./bin/mysql --socket=$POH_PATH/var/run/mysqld/mysql.sock -u root --port=$mysql_port -e "CREATE SCHEMA oh; GRANT ALL ON oh.* TO 'isf'@'localhost' IDENTIFIED BY 'isf123'; GRANT ALL ON oh.* TO 'isf'@'%' IDENTIFIED BY 'isf123';"
-    ./bin/mysql --socket=$POH_PATH/var/run/mysqld/mysql.sock -u root --port=$mysql_port oh < $POH_PATH/database.sql
+	echo "Starting MySQL Server."
+
+	$POH_PATH/$MYSQL_DIR/bin/mysqld_safe --defaults-file=$POH_PATH/etc/mysql/my.cnf 2>&1 > /dev/null &
+
+	# Wait till the MySQL socket file is created
+	while [ ! -e $POH_PATH/var/run/mysqld/mysql.sock ]; do sleep 1; done
+
+
+	echo "Dropping OH Database (if existing)..."
+	$POH_PATH/$MYSQL_DIR/bin/mysql --socket=$POH_PATH/var/run/mysqld/mysql.sock -u root --port=$MYSQL_PORT -e "DROP DATABASE IF EXISTS $DATABASE_NAME;"
+
+	echo "Creating OH Database..."
+	$POH_PATH/$MYSQL_DIR/bin/mysql --socket=$POH_PATH/var/run/mysqld/mysql.sock -u root --port=$MYSQL_PORT -e "CREATE DATABASE $DATABASE_NAME; GRANT ALL ON oh.* TO '$DATABASE_USER'@'localhost' IDENTIFIED BY '$DATABASE_PASSWORD'; GRANT ALL ON oh.* TO '$DATABASE_USER'@'%' IDENTIFIED BY '$DATABASE_PASSWORD';"
+
+	echo "Importing schema..."
+	$POH_PATH/$MYSQL_DIR/bin/mysql --socket=$POH_PATH/var/run/mysqld/mysql.sock -u root --port=$MYSQL_PORT oh < $POH_PATH/database.sql
     if [ $? -ne 0 ]; then
-		echo "Error: Database not initialized!"
+		echo -n "Error: Database not initialized!"
 		exit 2
     fi
-    echo "Database initialized."
-    rm $POH_PATH/database.sql
+    echo "Database initialized !"
+#    rm $POH_PATH/database.sql # we don't need to remove the database file
```

Comment:
we actually do need to remove that file: otherwise, next time you execute this script, it enters this if clause and it drops and recreates the database from scratch, **removing all persisted changes the user made** (!)

Reply:
Understood, but let me find a bettere way of dealing with it.
If for any reason I have to reinitiate the POH, I have to download or at least unzip everything from scratch.
Moreover, removing a file automatically distributed inside a packaget, from a launching script is never a good idea.

Reply:
The idea to allow the end-user to recreate the DB and starting from scratch is actually interesting... we could just rename the file instead of removing (or mv somewhere), and to preparare a script "oh_reset.sh" (aside oh.sh) that uses the so renamed (or moved) file to reinitialize the DB.

Reply:
good idea mwithi ! let me study a solution to propose.

Reply:
> we could just rename the file

yes that would be better

> preparare a script "oh_reset.sh" (aside oh.sh) that uses the so renamed (or moved) file to reinitialize the DB.

it may be useful (I don't know, I still have to meet OH users), but I'd rather consolidate existing features and scripts than add a new one (also, if we add it for Linux we must add it for Windows as well)

Reply:
Yes @pviotti I was thinking about the same "manutention" downside... 😃 probably a unique parametrized script would be better, but still need to be documented as well!

Let's just think what's the best solution and we'll put effort to that one

For the target users we found easier to distribute one script for each function, and the need to reset the database has actually been raised quite often (portable is for testing), but the words solution has always been "just delete and unzip again" (that is more understandable by any user in the same target)

Reply:
> otherwise, next time you execute this script, it enters this if clause and it drops and recreates the database from scratch, **removing all persisted changes the user made** (!)

just to be precise in this case the `--init` would fail and the application would exit, but no script would be restored so no user data would be lost

Reply:
Good, I will come up with a proposal trying to include all your suggestions.